js-src/src/app/map.component.js

```diff
@@ -17,167 +22,186 @@ let MapComponent = Component({
-                        self.initGame(self, game)
+                    self.gameService.getGame(game_id).subscribe(function (game) {
+                        self.initGame(self, game);
                     });
                 });
         },
+        createChannel: function (player_id) {
+            this.channelService.createChannel(player_id).subscribe(function (resp) {
+                console.log("channel_token:");
+                console.log(resp.token);
+                var channel = new goog.appengine.Channel(resp.token);
+                var handler = {
+                    'onopen': function () {
+                        console.log('opened channel');
+                    },
+                    'onmessage': function (msg) {
+                        console.log('received message: ' + msg.data);
+                    },
+                    'onerror': function () {
+                    },
+                    'onclose': function () {
+                    }
+                };
+                var socket = channel.open(handler);
+            });
+        },
         initGame: function (self, game) {
             self.game = game;
-			self.styleOfMap = self.noLabelAndBorder;
+            self.styleOfMap = self.noLabelAndBorder;
             self.markers = [];
-			self.cityMarkers = [];
-			self.infoWindows = [];
+            self.cityMarkers = [];
+            self.infoWindows = [];
             self.currentRound = 0;
             self.currentScore = 0;
-			self.startCountdown(0);//Initialize Round timer
+            // self.startCountdown(0);//Initialize Round timer
         },
         guessResponse: function (self, currentCity, guess) {
-            self.currentScore += guess.score;
-			if(!this.responseTaken){//If map is already clicked, dont let it be clicked again
-				this.roundTimer  = 0;//Set round timer to 0, round ended
-				self.currentScore += guess.score;
-				
-				// Changed self.markers.length to 2 as seperated markers and cityMarkers
-				if (self.currentRound < 2) {
-					this.responseTaken = true;
-					self.startCountdown(1);//Initialize Break timer
-					setTimeout(function () {
-						self.startCountdown(0);//Initialize Round timer
-						self.markers = [];
-						self.cityMarkers = [];
-						self.infoWindows = [];
-						self.currentRound++;
-					}, 3000);
-				} else {
-					this.gameEnded = true;//Game ended, do not initialize the counter for rounds
-					alert("Total Score: " + self.currentScore);
-					self.router.navigate(['/highscores', self.game.id, self.player.id ])//Move to highscores when the game ends
-				}
-			}
+            if (!this.responseTaken) {//If map is already clicked, dont let it be clicked again
+                this.roundTimer = 0;//Set round timer to 0, round ended
+                self.currentScore += guess.score;
+                // TODO: push different colored markers
+                self.markers.push({lat: currentCity.lat, lng: currentCity.long});
```

Comment:
What is this line of code used for? markers is used to push marker where player clicked; cityMarkers is to push the right marker where the city is.